core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -116,9 +116,14 @@ public static GetField fromXContent(XContentParser parser) throws IOException {
         String fieldName = parser.currentName();
         XContentParser.Token token = parser.nextToken();
         ensureExpectedToken(XContentParser.Token.START_ARRAY, token, parser::getTokenLocation);
+
         List<Object> values = new ArrayList<>();
         while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            values.add(parseStoredFieldsValue(parser));
+            if (token.isValue()) {
+                values.add(parseStoredFieldsValue(parser));
+            } else {
+                parser.skipChildren();
```

Comment:
Ok, I'm reverting this change. The parseStoredFieldsValue() is strict but not tested and it looks like it does not handle correctly the `null` value that can be printed out by MappedTypeField sub classes. I'm not sure it's a real case be we should test that too. I'll add a test for  parseStoredFieldsValue() as a follow up.